src/server/editorServices.ts

```diff
@@ -2722,8 +2781,21 @@ namespace ts.server {
                         else {
                             // reload from the disk
                             this.reloadConfiguredProject(project, reason);
+                            // If this is solution, reload the project till the reloaded project contains the script info directly
+                            if (!project.containsScriptInfo(info) && project.isSolution()) {
+                                forEachResolvedProjectReferenceProject(
+                                    project,
+                                    child => {
+                                        if (!updatedProjects.has(child.canonicalConfigFilePath)) {
+                                            updatedProjects.set(child.canonicalConfigFilePath, true);
+                                            this.reloadConfiguredProject(child, reason);
+                                        }
+                                        return projectContainsInfoDirectly(child, info);
+                                    },
+                                    ProjectReferenceProjectLoadKind.FindCreate

```

Comment:
Not `FindCreateLoad` because the callback's going to load it?

Reply:
Because we are reloading it anyways.. so createAndLoad is not needed